pkg/pipeline/utils/utils.go

```diff
@@ -262,3 +267,30 @@ func SetRegistryPortMapping(configmap *corev1.ConfigMap, portMap map[string]stri
 	configmap.Data[RegistryPortMappingFile] = string(b)
 	return nil
 }
+
+//EnsureAccessToken Checks expiry and do token refresh when needed
+func EnsureAccessToken(credentialInterface v3.SourceCodeCredentialInterface, remote model.Remote, credential *v3.SourceCodeCredential) (string, error) {
+	if credential == nil {
+		return "", nil
+	}
+	if refresher, ok := remote.(model.Refresher); ok {
```

Comment:
Nit: we can just turn right away if !ok, instead of wrapping the big chunk of a code in if statement when there is no else